agent/engine/engine_unix_integ_test.go

```diff
@@ -1381,3 +1381,98 @@ func TestHostResourceManagerStopTaskNotBlockWaitingTasks(t *testing.T) {
 
 	waitFinished(t, finished, testTimeout)
 }
+
+// Test Host Resource Manager does not account Fargate tasks when started
+func TestHostResourceManagerLaunchTypeBehavior(t *testing.T) {
+	testCases := []struct {
+		Name       string
+		LaunchType string
+	}{
+		{
+			Name:       "TestHostResourceManagerFargateLaunchTypeBehavior",
+			LaunchType: "FARGATE",
+		},
+		{
+			Name:       "TestHostResourceManagerEC2LaunchTypeBehavior",
+			LaunchType: "EC2",
+		},
+		{
+			Name:       "TestHostResourceManagerExternalLaunchTypeBehavior",
+			LaunchType: "EXTERNAL",
+		},
+		{
+			Name:       "TestHostResourceManagerRandomLaunchTypeBehavior",
+			LaunchType: "RaNdOmStrInG",
+		},
+		{
+			Name:       "TestHostResourceManagerEmptyLaunchTypeBehavior",
+			LaunchType: "",
+		},
+	}
+
+	for _, tc := range testCases {
+		testTimeout := 1 * time.Minute
+		taskEngine, done, _ := setupWithDefaultConfig(t)
+		defer done()
+
+		stateChangeEvents := taskEngine.StateChangeEvents()
+
+		taskArn := fmt.Sprintf("IntegTaskArn")
+		testTask := createTestTask(taskArn)
+		testTask.Memory = int64(768)
+		testTask.LaunchType = tc.LaunchType
+
+		// create container
+		A := createTestContainerWithImageAndName(baseImageForOS, fmt.Sprintf("A"))
```

Comment:
`Sprintf` is redundant here. Also can we please use a more descriptive name than `A`?